### PR TITLE
fix(Core/Loot): Properly save and load from DB loot from items.

### LINKED ISSUE #5472: [Item] Additional Items Possible From RDF Reward Box [$25]
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [ ] Alliance
- [x] Horde

##### CONTENT PHASE:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

Original report: https://github.com/chromiecraft/chromiecraft/issues/456

When receiving a Satchel of Helpful Goods from RDF run this usually contains 1 item. 

If you view the item, either by not having autoloot on, or shift right click then log off without emptying box, 2 more items appear in box. You can then loot these 3 items.

2021-04-23 14_30_13-World of Warcraft

WoWScrnShot_042621_113539

WoWScrnShot_042621_113552

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Box should only contain 1 item

##### STEPS TO REPRODUCE THE PROBLEM:

1. .additem [Satchel of Helpful Goods] 
2. View the contents without looting
3. Log off and back on
4. The satchel will now contain 2 additional bonus items.

##### EXTRA NOTES:

Tested on 2 chars in multiple different dungeons.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/b8b5f68f98ff8771a6ec87ea6805ceced91c406a

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
There is a **[$25 open bounty](https://www.bountysource.com/issues/98101603-item-additional-items-possible-from-rdf-reward-box?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #5472.

## Changes Proposed:
-  Added additional variables to DB.
-  Fixed loading not normal loot.

## Issues Addressed:

- Closes #5472
- Closes https://github.com/chromiecraft/chromiecraft/issues/456

## Tests Performed:

- Tested ingame with `Satchel Helpful Goods`

## How to Test the Changes:

1. `additem 52004`
2. Open the bag - do no loot.
3. Relog
4.. Open again the bag.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Loot/LootItemStorage.cpp [hunk 1]
```diff
@@ -134,24 +154,41 @@ bool LootItemStorage::LoadStoredLoot(Item* item)
...
+            li.randomSuffix = it2->randomSuffix;
+            li.rollWinnerGUID = ObjectGuid::Empty;
+
+            // Copy the extra loot conditions from the item in the loot template
+            lt->CopyConditions(&li);
+
+            if (li.needs_quest)
+                loot->quest_items.push_back(li);
+            else
+                loot->items.push_back(li);
```
Comment (FrancescoBorzi) [inline src/server/game/Loot/LootItemStorage.cpp:179]: please always use brackets

**Comment by tkn963** [comment]
Tested

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by eTzmNcbkrng** [comment]
This has added a regression where if you look inside a satchel without looting it then log out and back in, you can no longer loot the satchel.

**Comment by Sedatyf** [comment]
@eTzmNcbkrng do you have an issue in mind? Because I couldn't find any other issue about the problem you're talking about. Which means there isn't any regression as nobody reported it

**Comment by eTzmNcbkrng** [comment]
I'll log an issue for it. Is there an option in `worldserver.conf` to set number of players required for RDF or do I have to load up 5 clients?

**Comment by Azcobu** [comment]
@eTzmNcbkrng No need to run through RDF is there? .additem [Satchel of Helpful Goods] will get you all you need. 

Though also see https://github.com/azerothcore/azerothcore-wotlk/issues/5964 on testing RDF.

**Pull request merged**